shared/sub_plugin.go

```diff
@@ -0,0 +1,19 @@
+package shared
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+)
+
+type AttributedObservation[ObservationType any] struct {
+	OracleID    commontypes.OracleID
+	Observation ObservationType
+}
+
+type SubPlugin[QueryType any, ObservationType any, OutcomeType any] interface {
```

Comment:
Using `prevOutcome` instead of `outCtx`. We're mainly interested in the `prevOutcome` and it makes it easier to have all decoding on the top level, otherwise there might be cyclic dependencies or just complicating the code more.

Reply:
Please add these kinds of comments to the code so that they're persisted rather than a GH comment which is probably never read after a PR is merged.